internal/handlers/pg/msg_find.go

```diff
@@ -92,13 +92,16 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
 			fmt.Sprintf("collection name has invalid type %s", common.AliasFromType(collectionParam)),
 		)
 	}
-	// comment set through options.FindOne().SetComment() method
+
+	// get comment from options.FindOne().SetComment() method
 	if sp.comment, err = common.GetOptionalParam(document, "comment", sp.comment); err != nil {
 		return nil, err
 	}
-	// comment in query, e.g. db.collection.find({$comment: "test"})
-	if sp.comment, err = common.GetOptionalParam(filter, "$comment", sp.comment); err != nil {
-		return nil, err
+	// get comment from query, e.g. db.collection.find({$comment: "test"})
+	if filter != nil {
+		if sp.comment, err = common.GetOptionalParam(filter, "$comment", sp.comment); err != nil {
+			return nil, err
+		}
```

Comment:
Should we do the same in findAndModify?

Reply:
Not neccesary do it same in this PR.
This piece of code has been refactored due to a bug. The bug occurs when we do not provide a filter and at the same time try to get a value from it by key.
I can add `comment` in this PR or another. What do you think is the right thing to do?

Reply:
@ribaraka I think if you are about to divide PR either you don't close the current issue and add some comments to it or create a new issue with the tas (depends on situation and description)

Reply:
Good thought! The issue has been added.